src/DSManager.cpp

```diff
@@ -461,14 +467,19 @@ ReturnCode_t DiscoveryServerManager::deleteParticipant(
 
         std::lock_guard<std::recursive_mutex> lock(management_mutex);
 
+        if (participant == nullptr)
+        {
+            return ReturnCode_t::RETCODE_ERROR;
+        }
+
```

Comment:
I think this check should be done before taking the mutex as it is done in some of the other methods. This also applies to other methods where the check is done after taking the mutex instead of before.